version: '3'

networks:
  soot_site:


services:

  database:
    image: soot-site-database:latest
    container_name: database
    hostname: database
    command: ["postgres", "-c", "config_file=/etc/postgresql/postgresql.conf"]
    volumes:
      - ./volumes/db:/var/lib/postgresql/data
      - ./postgres/configs/:/configs/
    env_file:
      - ./postgres/configs/postgres.env
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "postgres", "-p", "5438"]
      interval: 15s
      timeout: 5s
      retries: 3
    networks:
      - soot_site

  backend:
    image: soot-site-backend:latest
    hostname: soot-api
    restart: always
    container_name: soot-api
    volumes:
      - ./backend:/app
    networks:
      - soot_site

  nginx:
    image: soot-site-nginx:latest
    container_name: soot-nginx
    restart: always
    ports:
      - "80:8080" # HTTP port
      - "1935:1935"  # RTMP port
    volumes:
      - ./nginx/configs/nginx-dev.conf:/etc/nginx/nginx.conf:ro
      - ./backend/logs/nginx:/var/log/nginx
      - ./backend/media/collectstatic/:/collectstatic/
      - ./backend/media/uploads/:/media/
      - ./nginx/tmp/hls:/tmp/hls
    networks:
      - soot_site
    depends_on:
      - backend